### DISCUSSION TITLE
use headtracking in place of smooth turning.

### DISCUSSION THREAD

**Discussion by Zod1234** [alvr-org/ALVR#1741] [General]
the main reason i use asw is for smooth turning in oculus with the right joystick as it can be choppy in some games with it off, but results in a bit of unsightly tearing when on.  moving through worlds isnt too bad, again it's mainly using smooth turning that makes fps issues really apparent.  

but simply looking around with the headset is smooth as glass in oculus, steamvr or alvr.  

so my idea is if left and right on the right joystick could be bound to headset translation.

**Comment by zmerp** [comment]
Interesting find 😅 but practically what you ask for is really hard to do. Smooth turning is a mechanic implemented by games, and the speed is unknown to ALVR or other drivers. And besides we also can't know when you intend to use the joystick for smooth turning or anything else. If it's really uncomfortable to look at you could enable back snap turning

**Reply by Zod1234** [reply]
well, as far as the game is concerned, you'd have to be able to disable turning altogether.  then if alvr had this option you'd enable it and set the rotation speed.  i also wondered would it feel more natural.  smooth turning usually rotates on a point, but we don't view the world that way, our view rotates around a center point, which headset rotation would mimic.

anyway, kind of far out idea, but i thought this would be a way to always have a high frame rate rotation.  thx for the quick reply.

**Reply by zmerp** [reply]
Good point. We could actually implement this in ALVR, and from the SteamVR point pf view the player is actually rotating inside the playspace. But this becomes a problem when using trackers and they would get misaligned.

**Comment by Zod1234** [comment]
if you do implement it please make it analog to be able to turn slowly when desired.  i've not got to play with trackers yet so hadn't considered that but i could see how that could cause issues.  it would be cool to see if it would work as I imagine.  Outside of trackers, I don't see why it wouldn't.  thx again.  loving v20.  finally have a layback option!

i also wonder if it turns out that it does feel more natural, it might be easier on some people that struggle a bit with smooth turning and are stuck with snap.

**Reply by zmerp** [reply]
Actually, the TrackingManager is implemented on the server, and so if we want implement smooth turning on the server as i described you would still see stuttering if the game/SteamVR is stuttering. So this requires more thought still.
